src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -125,15 +125,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // basicWithImages allows a, b, blockquote, br, cite, code, dd, dl, dt, em, i, li, ol,
     // p, pre, q, small, span, strike, strong, sub, sup, u, ul, and image tags
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.basicWithImages());
-    String recipient = request.getParameter("recipient");
+    String text  = Jsoup.clean(request.getParameter("text"), Whitelist.basicWithImages());
+    String recipient = request.getParameter("recipient");   
+      
+ 
+    String regex = "(https?://([^\\s.]+.?[^\\s.])+/([^\\s.]+.?[^\\s.])+.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    
+    String textWithImagesReplaced = text.replaceAll(regex, replacement);
```

Comment:
Alright. Lets be careful here. You are using `textWithImagesReplaced` here but below (line 139) you are using `text`. Also you can remove the `System.out.println` in the next line.